### PR TITLE
implement read/return buttons under My Loans

### LINKED ISSUE #7443: Implement new read/return buttons on my books page
Implement @danafein 's mockups:

In scope:

- Add return button of style in mockups
- Change colour of expiry date
- Make expiry date relative using the `datestr(date)` function.

### Stakeholders
@danafein @jimchamp @mekarpeles

### PR DESCRIPTION
**Pull request by JaydenTeoh**
Closes #7443 

Implemented Read/Return buttons for my loans

### Testing

Will need to test whether buttons are functioning properly on staging environment as borrowing API is not enabled for local environment.
### Screenshot

Return book

### Stakeholders

@cdrini created issue
@mekarpeles

### CODE REVIEW

File: openlibrary/macros/FormatExpiry.html [hunk 1]
```diff
@@ -19,6 +19,9 @@
       $_('Expires')
       <span class="localize-time"
 	    data-expiry-utc="$datetimestr_utc(expiry_datetime)">
-	$datetimestr_utc(expiry_datetime)
+        $if relative:
+            $datestr(expiry_datetime)
+        $else:
+            $datetimestr_utc(expiry_datetime)
```
[the remark this answers is not in the document]
Comment (cdrini) [reply openlibrary/macros/FormatExpiry.html:25]: NOTE: I'm not sure if the timezones here are/were ever correct 😬 

Is the datetime in eg https://openlibrary.org/account/loans.json UTC? CC @mekarpeles 

The datetime from the loans endpoint (eg something like `"2023-03-10 04:42:11"`) is at some point parsed to be a `datetime` object. Then this object is passed through `datetimestr_utc`. This adds a `Z` at the end of the datetime, but does not change the time in any way; and it adds e.g. `T` between to put it in proper ISO format.

The JS then converts this string using `toLocaleString`. This _changes the time_, but it appears to change the time incorrectly.

1. The initial `expiry` was listed as `2023-03-10 04:42:11`
2. After `datetimestr_utc` it is `2023-03-15T04:42:11Z`
3. After the JS, it is `03/15/2023, 12:42 AM` for me. Note this time conversion is wrong for my timezone. 04:42 UTC is 23:42

File: openlibrary/templates/books/custom_carousel.html [hunk 2]
```diff
@@ -60,7 +60,14 @@
       </a>
     </div>
     <div class="book-cta">
-      $:macros.LoanStatus(book, work_key=work_key, listen=False)
+      $if check_loan_status:
```
[the remark this answers is not in the document]
Comment (cdrini) [reply openlibrary/templates/books/custom_carousel.html:63]: Has different style:
image

Let's use it anyways, and change the style on the books page to put the date under the return button as well.

File: openlibrary/macros/FormatExpiry.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (cdrini) [reply openlibrary/macros/FormatExpiry.html:25]: Sigh I think the are some questions around how to make the timezone correct; so to keep things simple, let's roll this back and leave it as it was before. I'll create a new issue to investigate making these relative with the timezone correct.

**Review by cdrini — CHANGES_REQUESTED** [review]

**Comment by cdrini** [comment]
If it's useful for testing, you can use this to get some stub loan data:

lending.py:497

```py
def get_loans_of_user(user_key):
    """TODO: Remove inclusion of local data; should only come from IA"""
    return list(map(Loan, [{"_key": "loan-styledsecretsfor0000hend", "_rev": 1, "type": "/type/loan", "userid": "@drini776", "user": "/people/ScarTissue", "book": "/books/OL2058361M", "ocaid": "styledsecretsfor0000hend", "expiry": datetime.datetime.fromisoformat("2023-03-10 04:42:11"), "fulfilled": True, "uuid": "loan-2731207935", "loaned_at": 1677645731.0, "resource_type": "bookreader", "resource_id": "bookreader:styledsecretsfor0000hend", "loan_link": None, "stored_at": "ia"}]))
```

**Comment by JaydenTeoh** [comment]
@cdrini I have made the changes requested.
1. I have used `LoanStatus.html` component for rendering the read/return buttons
2. I have reversed the order of the return button and expiry date in `LoanStatus.html`
3. I rollbacked the changes to `FormatExpiry.html`

Here is how a loan in the loans carousel of mybooks page look like:
My Loans (1)

I am struggling to stub a loan book on the books page to see how the new `databarWork.html` component will look like, could you guide me through it?

File: openlibrary/macros/FormatExpiry.html [hunk 3]
```diff
@@ -19,6 +19,6 @@
       $_('Expires')
       <span class="localize-time"
 	    data-expiry-utc="$datetimestr_utc(expiry_datetime)">
-	$datetimestr_utc(expiry_datetime)
+            $datetimestr_utc(expiry_datetime)
```
Comment (cdrini) [inline openlibrary/macros/FormatExpiry.html:22]:
```suggestion
	$datetimestr_utc(expiry_datetime)
```

**Review by cdrini — APPROVED** [review]
Lgtm! Tested buttons are now there and work. As followups we should

1) Investigate making the date relative. Note the commend
2) Fix spacing on work page; looks funky.
3) After returning, don't go to /loans page. Return to referrer page.

**Comment by cdrini** [comment]
@JaydenTeoh getting it in databarWork is a bit more difficult! I've put this up on testing.openlibrary.org and it is working well :)

**Pull request merged**